ogr/ogrsf_frmts/miramon/mm_gdal_functions.c

```diff
@@ -0,0 +1,2801 @@
+        {
+            for (j = 0; j < MM_NUM_IDIOMES_MD_MULTIDIOMA; j++)
+            {
+                cadena = data_base_XP->pField[i].Separator;
+                if (cadena[j])
+                {
+                    free_function(cadena[j]);
+                    cadena[j] = nullptr;
+                }
+            }
+        }
+        free_function(data_base_XP->pField);
+        data_base_XP->pField = nullptr;
+        data_base_XP->nFields = 0;
+    }
+    return;
+}
+
+// READING THE HEADER OF AN EXTENDED DBF
+// Free with MM_ReleaseDBFHeader()
+int MM_ReadExtendedDBFHeaderFromFile(const char *szFileName,
+                                     struct MM_DATA_BASE_XP *pMMBDXP,
+                                     const char *pszRelFile)
+{
+    MM_BYTE variable_byte;
+    FILE_TYPE *pf;
+    unsigned short int ushort;
+    MM_EXT_DBF_N_FIELDS nIField;
+    MM_FIRST_RECORD_OFFSET_TYPE offset_primera_fitxa;
+    MM_FIRST_RECORD_OFFSET_TYPE offset_fals = 0;
+    MM_BOOLEAN incoherent_record_size = FALSE;
+    MM_BYTE un_byte;
+    MM_BYTES_PER_FIELD_TYPE_DBF bytes_per_camp;
+    MM_BYTE tretze_bytes[13];
+    MM_FIRST_RECORD_OFFSET_TYPE offset_possible;
+    MM_BYTE n_queixes_estructura_incorrecta = 0;
+    MM_FILE_OFFSET offset_reintent = 0;  // For retrying
+    char cpg_file[MM_CPL_PATH_BUF_SIZE];
+    char *pszDesc;
+    char section[MM_MAX_LON_FIELD_NAME_DBF + 25];  // TAULA_PRINCIPAL:field_name
+    GUInt32 nRecords;
+    char *pszString;
+
+    if (!szFileName)
+        return 1;
+
+    strcpy(pMMBDXP->szFileName, szFileName);
+    strcpy(pMMBDXP->ReadingMode, "rb");
+
+    if ((pMMBDXP->pfDataBase = fopen_function(pMMBDXP->szFileName,
+                                              pMMBDXP->ReadingMode)) == nullptr)
+        return 1;
+
+    pf = pMMBDXP->pfDataBase;
+
+    fseek_function(pf, 0, SEEK_SET);
+    /* ====== Header reading (32 bytes) =================== */
+    offset_primera_fitxa = 0;
+
+    if (1 != fread_function(&(pMMBDXP->dbf_version), 1, 1, pf) ||
+        1 != fread_function(&variable_byte, 1, 1, pf) ||
+        1 != fread_function(&(pMMBDXP->month), 1, 1, pf) ||
+        1 != fread_function(&(pMMBDXP->day), 1, 1, pf))
+    {
+        fclose_function(pf);
+        return 1;
+    }
+
+    if (1 != fread_function(&nRecords, 4, 1, pf))
+    {
+        fclose_function(pf);
+        return 1;
+    }
+
+    if (1 != fread_function(&offset_primera_fitxa, 2, 1, pf))
+    {
+        fclose_function(pf);
+        return 1;
+    }
+
+    pMMBDXP->year = (short)(1900 + variable_byte);
+reintenta_lectura_per_si_error_CreaCampBD_XP:
+
+    if (n_queixes_estructura_incorrecta > 0)
+    {
+        if (!MM_ES_DBF_ESTESA(pMMBDXP->dbf_version))
+        {
+            offset_fals = offset_primera_fitxa;
+            if ((offset_primera_fitxa - 1) % 32)
+            {
+                for (offset_fals = (offset_primera_fitxa - 1);
+                     !((offset_fals - 1) % 32); offset_fals--)
+                    ;
+            }
```

Comment:
my understanding is that you want to clear the 5 lowest bits to get a multiple of 32, right ? If so, the following should do:
```suggestion
            offset_fals = offset_primera_fitxa & (MM_FIRST_RECORD_OFFSET_TYPE)(~31);
```
Cf
```
>>> 32 & ~31
32
>>> 33 & ~31
32
>>> 63 & ~31
32
>>> 64 & ~31
64
```